File: Leetcode_Python/max-operations.py
class Solution(object):
    def maxOperations(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        pointer1 = 0
        pointer2 = len(nums)-1
        count = 0

        nums.sort()

        while pointer2 > pointer1:
            if nums[pointer2] + nums[pointer1] > k:
                pointer2 -= 1
            elif nums[pointer2] + nums[pointer1] < k:
                pointer1 += 1
            else:
                count += 1
                pointer1 += 1
                pointer2 -= 1
        return count

File: Leetcode_TS/max-operations.ts
function maxOperations(nums: number[], k: number): number {
    let pointer1: number = 0
    let pointer2: number = nums.length-1
    let count:number = 0

    nums.sort((a,b) => a-b)

    while (pointer2 > pointer1) {
        if (nums[pointer1] + nums[pointer2] === k) {
            count++
            pointer1++
            pointer2--
        } else if (nums[pointer1] + nums[pointer2] > k) {
            pointer2--
        } else {
            pointer1++
        }
    }
    return count
};